File: scala/BeginnersSelection/06_CardGameForTwo.scala
// https://atcoder.jp/contests/abs/tasks/abc088_b

object Main {

  implicit class Pipeline[T](x: T) {
    def |>[S](f: T => S): S = f(x)
  }

  def main(args: Array[String]) = (
    io.StdIn.readLine
      |> (_ => io.StdIn.readLine
        .split(" ")
        .map(_.toInt)
        .sorted
        .reverse
        .zipWithIndex
        .fold((0, 0)) { case ((a, b), (x, i)) =>
          if (i % 2 == 0)
            (a + x, b)
          else
            (a, b + x)
        }
      )
      |> { case (a, b) => a - b }
      |> println
  )
}

// https://atcoder.jp/contests/abs/submissions/19772466

File: py/BeginnersSelection/06_CardGameForTwo.py
# https://atcoder.jp/contests/abs/tasks/abc088_b

from functools import reduce

class W:
  def __init__(self, v): self.v = v
  let = lambda self, f: W(f(self.v))
  map = lambda self, f: W(map(f, self.v))
  fold = lambda self, f, init: W(reduce(f, self.v, init))

W([input(), input()][1].split()) \
  .map(int) \
  .let(lambda x: sorted(x, reverse=True)) \
  .fold(lambda t, x: (
    lambda a, b, i:
      (a + x, b, i + 1)
      if i % 2 == 0 else
      (a, b + x, i + 1)
  )(t[0], t[1], t[2]), (0, 0, 0)) \
  .let(lambda t: t[0] - t[1]) \
  .let(print)

# https://atcoder.jp/contests/abs/submissions/21717312
